op-e2e/Makefile

```diff
@@ -1,4 +1,4 @@
-num_cores := $(shell nproc)
+num_cores := $(shell nproc 2>/dev/null || sysctl -n hw.ncpu 2>/dev/null || echo 4)
```

Comment:
Probably worth adding a comment here to explain that nproc is for linux, sysctl for Mac and then fallback to 4 if neither is available.